packages/gestalt/src/InternalSheet.js

```diff
@@ -28,10 +29,36 @@ type InternalSheetProps = {|
   children: NodeOrRenderProp,
   closeOnOutsideClick: boolean,
   footer: NodeOrRenderProp,
-  heading: NodeOrRenderProp,
+  enabledDismiss: boolean,
+  heading?: string,
+  message?: string,
+  message?: string,
```

Comment:
It looks like `message` got duplicated here.

Also: I wouldn't recommend spreading these props yet. To InternalSheet, the abstraction of `dismissConfirmation` as an object makes sense, as that's just a small part of what InternalSheet does (and really InternalSheet doesn't do anything with the prop aside from pass it along). But it does make sense to spread the `dismissConfirmation` fields to SheetConfirmationPopover, since that component does care about the individual fields. So something like
```js
// Sheet.js
type Props = {|
  dismissConfirmation: {| message, primaryAction, etc |}
|};
...
<InternalSheet dismissConfirmation={dismissConfirmation} />
...
```
```js
// InternalSheet.js
type Props = {|
  dismissConfirmation: {| message, primaryAction, etc |}
|};
...
<SheetConfirmationPopover {...dismissConfirmation} />
...

```
```js
// SheetConfirmationPopover.js
type Props = {|
  message: string,
  primaryAction: {| .. |},
  etc
|}
```
Does that make sense? Like continue to use the object abstraction as a way of keeping the related props together through Sheet and InternalSheet, then spread them where they're being used (SheetConfirmationPopover) and the object abstraction is no longer necessary

Reply:
DONE/UNDONE